libbeat/tests/integration/framework.go

```diff
@@ -275,12 +270,25 @@ func (b *BeatProc) waitBeatToExit() int {
 func (b *BeatProc) Stop() {
 	b.cmdMutex.Lock()
 	defer b.cmdMutex.Unlock()
+	b.stopUnsynced()
+}
+
+// stopUnsynced is the actual stop code, but without locking so it can be reused
+// by methods that have already acquired the lock.
+func (b *BeatProc) stopUnsynced() {
+	stopTimeout := 5 * time.Second
+
 	if err := b.Process.Signal(os.Interrupt); err != nil {
 		if errors.Is(err, os.ErrProcessDone) {
 			return
 		}
 		b.t.Fatalf("could not stop process with PID: %d, err: %s", b.Process.Pid, err)
 	}
+
+	stoppedLog := fmt.Sprintf("%s stopped.", b.beatName)
+	b.WaitForLogs(fmt.Sprintf("%s stopped.", b.beatName), stopTimeout,
```

Comment:
You're generating the same string twice:
```suggestion
	stoppedLog := fmt.Sprintf("%s stopped.", b.beatName)
	b.WaitForLogs(stoppedLog, stopTimeout,
```